cai_causal_graph/time_series_causal_graph.py

```diff
@@ -520,57 +650,53 @@ def add_edge(
         In addition to the `CausalGraph.add_edge` method, this method also populates the metadata of the nodes with
         the variable name and the time lag.
         """
+        # if the source and destination time series nodes do not exist, create them
+        if source is not None and not self.node_exists(source):
+            source_meta = None
+            if isinstance(source, HasMetadata):
+                source_meta = source.get_metadata()
+                source = self.add_node(source, meta=source_meta)
+
+        if destination is not None and not self.node_exists(destination):
+            destination_meta = None
+            if isinstance(destination, HasMetadata):
+                destination_meta = destination.get_metadata()
+                destination = self.add_node(destination, meta=destination_meta)
+
         edge = super().add_edge(source, destination, edge_type=edge_type, meta=meta, edge=edge, **kwargs)
-        # populate the metadata for each node
-        for node in self.get_nodes():
-            # extract variable name and lag from node name
-            vname, lag = get_variable_name_and_lag(node.identifier)
-            TimeSeriesCausalGraph._update_node_meta(node, vname, lag)
 
         return edge
 
-    @classmethod
-    def from_causal_graph(cls, causal_graph: CausalGraph, store_graphs: bool = False) -> TimeSeriesCausalGraph:
+    @staticmethod
+    def from_causal_graph(causal_graph: CausalGraph) -> TimeSeriesCausalGraph:
```

Comment:
This is a @classmethod in the super class right? I think I would leave as @classmethod then even if you don't use `cls`